cmd/apichecker/apicheck.go

```diff
@@ -0,0 +1,1344 @@
+		if err != nil {
+			return "", err
+		}
+		y, err := time.Parse(htmlData.DateFormat, dataStrings[len(dataStrings)-1])
+		if err != nil {
+			return "", err
+		}
+		z := y.Sub(x)
+		switch {
+		case z > 0:
+			return dataStrings[len(dataStrings)-1], nil
+		case z < 0:
+			return dataStrings[0], nil
+		default:
+			return "", errors.New("two or more updates were done on the same day, please check manually")
+		}
+	default:
+	}
+	return "", errors.New("no response found")
+}
+
+// Add appends exchange data to updates.json for future api checks
+func Add(fileName, exchName, checkType, path string, data interface{}, update bool) error {
+	finalResp, check, err := CheckExistingExchanges(jsonFile, exchName)
+	if err != nil {
+		return err
+	}
+	var file []byte
+	switch update {
+	case false:
+		if check {
+			if verbose {
+				log.Printf("%v exchange already exists\n", exchName)
+			}
+			return nil
+		}
+		exchange, err := FillData(exchName, checkType, path, data)
+		if err != nil {
+			return err
+		}
+		finalResp = append(finalResp, exchange)
+		file, err = json.MarshalIndent(finalResp, "", " ")
+		if err != nil {
+			return err
+		}
+	case true:
+		info, err := FillData(exchName, checkType, path, data)
+		if err != nil {
+			return err
+		}
+		allExchData, err := Update(exchName, finalResp, info)
+		if err != nil {
+			return err
+		}
+		file, err = json.MarshalIndent(allExchData, "", " ")
+		if err != nil {
+			return err
+		}
+	}
+	return ioutil.WriteFile(jsonFile, file, 0770)
+}
+
+// FillData fills exchange data based on the given checkType
+func FillData(exchName, checkType, path string, data interface{}) (ExchangeInfo, error) {
+	switch checkType {
+	case github:
+		tempSha, err := getSha(path)
+		if err != nil {
+			return ExchangeInfo{}, err
+		}
+		return ExchangeInfo{
+			Name:      exchName,
+			CheckType: checkType,
+			Data: &CheckData{
+				GitHubData: &GithubData{
+					Repo: path,
+					Sha:  tempSha.ShaResp},
+			},
+		}, nil
+	case htmlScrape:
+		tempData := data.(HTMLScrapingData)
+		checkStr, err := CheckChangeLog(tempData)
+		if err != nil {
+			return ExchangeInfo{}, err
+		}
+		return ExchangeInfo{
+			Name:      exchName,
+			CheckType: checkType,
+			Data: &CheckData{
+				HTMLData: &HTMLScrapingData{
+					CheckString:   checkStr,
+					DateFormat:    tempData.DateFormat,
+					Key:           tempData.Key,
+					RegExp:        tempData.RegExp,
+					TextTokenData: tempData.TextTokenData,
+					TokenData:     tempData.TokenData,
+					TokenDataEnd:  tempData.TokenDataEnd,
+					Val:           tempData.Val,
+					Path:          tempData.Path},
+			},
+		}, nil
+	default:
+		return ExchangeInfo{}, errors.New("invalid checkType")
+	}
+}
+
+// HTMLScrapeDefault gets check string data for the default cases
+func HTMLScrapeDefault(htmlData HTMLScrapingData) ([]string, error) {
+	var resp []string
+	temp, err := http.Get(htmlData.Path)
+	if err != nil {
+		return resp, err
+	}
+	tokenizer := html.NewTokenizer(temp.Body)
+loop:
```

Comment:
The only time you use this is in `case html.ErrorToken` so you could get rid of `loop:` and for `case html.ErrorToken:` just use continue